tests/test_networks.py

```diff
@@ -858,3 +859,33 @@ class Model(BaseModel):
     assert exc_info.value.errors() == [
         {'input': 1, 'loc': ('v',), 'msg': 'Input is not a valid NameEmail', 'type': 'name_email_type'}
     ]
+
+
+@pytest.mark.skipif(not email_validator, reason='email_validator not installed')
+def test_name_email_json():
+    class Model(BaseModel):
+        v: NameEmail
+
+    assert Model.model_validate_json('{"v":{"name":"foo bar","email":"foobaR@example.com"}}').v == NameEmail(
+        'foo bar', 'foobaR@example.com'
+    )
+
+    assert Model.model_validate_json('{"v":{"name":"foo bar","email":"whatever <foobaR@example.com>"}}').v == NameEmail(
```

Comment:
IDK if this case should be allowed or should raise an exception. Coz the implementation just override the name from dict and throw away parsed name.

Reply:
I don't think we want to support this case, as mentioned in my above comment.